components/server/src/express-util.ts

```diff
@@ -91,7 +91,14 @@ const looksLikeWorkspaceHostname = (originHostname: URL, gitpodHostName: string)
         return false;
     }
     const url = new GitpodHostUrl(originHostname);
-    return workspaceIDRegex.test(url.workspaceId || '')
+    const workspaceId = url.workspaceId;
+    if (workspaceId) {
+        const hostname = url.url.hostname as string;
+        if (hostname.startsWith(workspaceId)) {
```

Comment:
Would be nice if we had one place for the workspace URL parsing. But might be out of scope for this PR, as it was here before, too.

Reply:
`GitpodHostUrl` does the URL parsing. Here we explicitly check, that the workspace-like hostname is not prefixed.
Ah, perhaps you mean to extract the computation of the prefix and move it to `GitpodHostUrl`?  🤷🏻‍♂️